java/src/main/java/science/icebreaker/network/keyword_merge/comparators/SimilarityResult.java

```diff
@@ -6,16 +6,24 @@
  * Represents the result of comparing 2 keywords for similarity
  */
 public class SimilarityResult {
+    public final Keyword kw1;
+    public final Keyword kw2;
     // Useful for deciding a common root keyword(might come in handy when inserting new data)
     public final Keyword origin;
     public final Boolean isSimilar;
 
-    public SimilarityResult(Keyword origin, Boolean isSimilar) {
+    public SimilarityResult(Keyword kw1, Keyword kw2, Keyword origin, Boolean isSimilar) {
+        this.kw1 = kw1;
+        this.kw2 = kw2;
         this.origin = origin;
         this.isSimilar = isSimilar;
     }
 
     public boolean equals(SimilarityResult result) {
-        return origin.equals(result.origin) && (this.isSimilar == result.isSimilar);
+        return 
+            kw1.equals(result.kw1)
```

Comment:
Is this going to work correctly since `Keyword.equals()` has not been overwritten? And is it intended that you are not overwriting `public boolean equals(Object o)` here?

Reply:
If i remember correctly, similar keywords have the same String, but yes, the equals should be implemented either way. I'll take a look at it

Reply:
This could have caused problems in generating the CSVs. luckily, these were just strings back then. I changed it anyway